File: src/test.checks/src/credits/models/Vendor.cs
//-----------------------------------------------------------------------------
// Copyright   :  (c) Chris Moore, 2020
// License     :  MIT
//-----------------------------------------------------------------------------
namespace Z0
{
    partial class CreditModel
    {
        /// <summary>
        /// Defines vendor reference component values
        /// </summary>
        public enum Vendor : byte
        {
            None = 0,

            /// <summary>
            /// Intel documentation
            /// </summary>
            Intel = 1,

            /// <summary>
            /// AMD documentation
            /// </summary>
            Amd = 2
        }
    }
}

File: src/tools/site/core/vendor.ts
import { Root } from "./root"

export type VendorKey = 
    | 'ms'
    | 'intel'
    
export type Vendor<K extends VendorKey,B> = Root<'vendor',B>


export function vendor<K extends VendorKey,B>(key:K, base:B) {
    return {
        key,
        base
    }
}
